### ISSUE TITLE
using page.setContent in 2.x.x

### ISSUE THREAD

**Issue by mcwu** [amir20/phantomjs-node#540]
Hi,

I am trying to use phantomjs-node to open a page, set the url, but load my own html.  This way, if the html had any relative urls, it would use the url that I set as the base.   I used to be able to do this using page.setContent(html, url, function() {}) .  However, I have not found any examples online for doing the same thing in version 2.0.0 and above using the then() pattern.  Can someone point me in the right direction?

Thanks!

**Comment by amir20** [comment]
It should still work. What did you try? setContent just converted to using a promise. I don't have any examples handy. 

Sent via my iPhone

**Comment by mcwu** [comment]
Ah, I figured out the issue.  I had something along the lines of:

var myPage;
phantomInstance.createPage.then(function(page) {
  myPage = page;
  myPage.open('about:blank');
}).then(function() {
  myPage.setContent(myHTML, myURL);
}).then(function() {
  myPage.renderBase64('JPEG');
}).then(function(image) {
  ...
});

and the image ended up being blank.  However I learned from this post (https://github.com/amir20/phantomjs-node/issues/68)  that it's because if there is external resources in the html, I need to wait for the external resources to finish loading before rendering.

Thanks for your help!
